### DISCUSSION TITLE
Best practice to preserve 2-manifold over epeck -> epick -> epeck roundtrips?

### DISCUSSION THREAD

**Discussion by pentacular** [CGAL/cgal#7521] [Q&A]
I've been using alpha_wrap_3  which I believe produces a 2-manifold.

However, when using an epick kernel, this can result in duplicate vertices, edges, etc, making the result not be a proper 2-manifold anymore.

Is there a best practice to repair this case?

And in general to handle an epeck -> epick -> epeck round trip?

**Comment by MaelRL** [comment]
By default, Alpha_wrap_3 is supposed to produce a 2-manifold without duplication or self-intersections. Please open an issue with your code and data input if this isn't the case.

**Comment by sloriot** [comment]
Did I understand correctly that you are using Alpha_wrap_3 with EPECK?

**Comment by pentacular** [comment]
Sorry, I think I expressed myself confusingly.

I mean to say that I am converting epeck to epick to use alpha_wrap_3 then converting the result of that back to epeck.

I then feed it into approximate_triangle_mesh where is_polygon_soup_a_polygon_mesh fails due to finding a duplicate edge:
```      if(! edge_set.insert(std::pair<V_ID, V_ID>(prev,id)).second) { ```

My assumption was that the epick output had become non-manifold due to points merging, which would be an expected problem, and have some recommended way to handle it.

Given that this sounds like a bug, I will try to produce a minimal test case and get back to you.

Thanks.
